proxy/http2/Http2ConnectionState.cc

```diff
@@ -1679,6 +1673,12 @@ Http2ConnectionState::send_a_data_frame(Http2Stream *stream, size_t &payload_len
     payload_length = 0;
   }
 
+  if (payload_length > 0 && this->session->is_write_high_water()) {
```

Comment:
You are not going to send data to the client unless there is 32k to send?

Reply:
If you're wondering about the timing of the `this->session->flush()`, it is called with several conditions. 
- e.g. if the response body is 10 bytes, it'll be sent to the client immediately.

This is checking the high watermark of the session buffer to stop it keep growing.